ui/pom.xml

```diff
@@ -152,4 +146,63 @@
       </plugin>
     </plugins>
   </build>
+  
+  <profiles>
+    <profile>
+      <id>build-docker-image</id>
+      <activation>
+        <property>
+          <name>env.DOCKER_HOST</name>
+        </property>
+      </activation>
+      <build>
+        <plugins>
+          <plugin>
+            <artifactId>maven-antrun-plugin</artifactId>
+            <version>1.8</version>
+            <executions>
+              <execution>
+                <phase>package</phase>
+                <goals>
+                  <goal>run</goal>
+                </goals>
+              </execution>
+            </executions>
+            <configuration>
+              <target>
+                <sync todir="${basedir}/docker/dist">
```

Comment:
Could we use `${project.build.directory}/docker/dist` instead of copying over into a top-level dir ?  This would then end up in `target/` and hence also gets cleaned up with a  `mvn clean`. I'd like to avoid mixing directories with committed (source) files and generated files (dist). 

For example, it's hard to clean up such a setup and if e.g. a files gets removed which should not appear in dist/ but you already had a run before then you end up with the old files still in `docker/dist`.  The ant sync taks takes care of that, but I'd still would prefer the 'mvn way' (i.e. working in target, with `mvn clean` giving you a fresh state).

Reply:
Of course one would need to copy over the Dockerfile + helper files then, too, to target, before calling docker.

Reply:
Yeah not a bad idea.  Isn't the a src/main/docker dir for that?

Reply:
yeah, we can put the Dockerfile into this directory. But we then still have to copy over these files + dist/ into a common directory below `target/` and then refer `<dockerFile>` to this merged directory.

I can optimize this later (we can use an assembly also together with a Dockerfile) but I would do this then when moving to fabric8-maven-plugin (not sure what version fabric8-maven-plugin uses which docker-maven-plugin).